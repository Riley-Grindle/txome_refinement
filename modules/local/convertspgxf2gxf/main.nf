
process AGAT_CONVERTSPGXF2GXF {
    tag "$meta.id"
    label 'process_high'
    
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/agat:1.0.0--pl5321hdfd78af_0' :
        'biocontainers/agat:1.4.0--pl5321hdfd78af_0' }"

    input:
    tuple val(meta), path(gtf)

    output:
    path("*.agat.gtf")  , emit: output_gtf
    path("*.log")       , emit: log
    path "versions.yml" , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    agat config --expose
    
    sed -i.bak 's/output_format: *.*/output_format: GTF/' agat_config.yaml
    rm *.bak
    
    agat_convert_sp_gxf2gxf.pl \\
        -g $gtf \\
        -o ${prefix}.agat.gtf \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        agat: \$(agat_convert_sp_gxf2gxf.pl --help | sed '4!d; s/.*v//')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.agat.gff
    touch ${gff}.agat.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        agat: \$(agat_convert_sp_gxf2gxf.pl --help | sed '4!d; s/.*v//')
    END_VERSIONS
    """
}

