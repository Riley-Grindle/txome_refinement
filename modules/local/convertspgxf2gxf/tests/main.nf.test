nextflow_process {

    name "Test Process AGAT_CONVERTSPGXF2GXF"
    script "../main.nf"
    process "AGAT_CONVERTSPGXF2GXF"

    setup {
        process {
            """
            input[0] = Channel.of( [ [ id: "TEST"], file(params.gtf, checkIfExists: true ) ]  ) 
            """
        }
    }

    test("GTF Format | AGAT | TEST") {

        when {
            process {
                """
                input[0] = Channel.of( [ [ id: "TEST"], file(params.gtf, checkIfExists: true ) ]  ) 
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )    
        }

}
}