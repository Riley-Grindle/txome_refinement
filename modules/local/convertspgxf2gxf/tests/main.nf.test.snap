{
    "GTF Format | AGAT | TEST": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "final_gtf": [
                    
                ],
                "log": [
                    
                ],
                "output_gtf": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-12-04T16:59:25.47287"
    }
}